course/include/isometry.h

```diff
@@ -0,0 +1,83 @@
+#ifndef ISOMETRY_H
+#define	ISOMETRY_H
+
+// Standard libraries
+#include <cmath>
+#include <initializer_list>
+#include <iostream>
+#include <limits>
+#include <sstream>
+#include <string>
+
+namespace ekumen {
+namespace math {
+
+static constexpr int resolution{4};
+template <class T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, bool>::type almost_equal(T a, T b, int resolution) {
+    return std::fabs(a - b) <= std::numeric_limits<T>::epsilon() * std::fabs(a + b) * resolution
+           || std::fabs(a - b) < std::numeric_limits<T>::min();
+}
+
+class Vector3 {
+    public:
+        Vector3(const double &x = 0, const double &y = 0, const double &z = 0);
```

Comment:
Add copy and move constructors.

Reply:
Make the Vector3 class movable, and follow the rule of 5 both in Vector3 and in Matrix3.

For the sake of the exercise make `Vector3` something that moving from is worth it. I suggest instead of storing the elements as "double x_,y_,z_;" store them in a struct:
```
struct Elements {
  double x_,y_,z_;
};
```
then in `Vector3`, declare the storage as:
```
private:
   Elements *v_;
```
such that you allocate the space for `v_` when you construct with `new` and deallocate it with `delete` in the destructor (if needed, beware when you move from an object).

The rest of the changes should be trivial, `x_` will become `v_->x`, `y_` will become`v_->y` and `z_` will become `v_->z`, and your test coverage will protect you against messing up during refactoring.

Again, make sure you follow the rule of five in `Vector3`, and then do the same in `Matrix3`. If you do it right, you'll see  no differences... if you mess the move copy operations you may have memory access problems and premature death of the test.

Notice you should add test cases for your move operations , to make sure your move operations are getting selected when you think they are; there are no hard rules for how to test that, but in your case you can add a "bool moved()" method that returns `true` if the Vector3 object is the result of a move operation/constructor, and `false` in other cases. You can then test its value in the resulting object after a copy/move test.